docs/development/journalist_api.rst

```diff
@@ -0,0 +1,414 @@
+Journalist Interface API
+========================
+
+This document describes the endpoints for SecureDrop's Journalist Interface
+API.
+
+Versioning
+~~~~~~~~~~
+
+The API is versioned and we are currently using version 1. This is set via the
+base URL, which is:
+
+.. code:: sh
+
+  /api/v1/
+
+Content Type
+~~~~~~~~~~~~
+
+Clients should send the following headers:
+
+.. code:: sh
+
+  'Accept': 'application/json',
+  'Content-Type': 'application/json'
+
+Authentication
+~~~~~~~~~~~~~~
+
+``POST /api/v1/token`` to get a token with the username, password, and 2FA
```

Comment:
Another note: HOTP is a PITA for this. We need to be able to renew tokens so that a client can authorize via HTOP once and then have an uninterrupted workflow. I think this is not a "nice to have" and needs to be implemented for this PR.

Reply:
Pros of implementing this: Users will otherwise need to login again when the token expires (bad UX)

Cons of implementing this: we will be allowing clients to potentially be authenticated forever (if the client keeps getting refresh tokens and they don't let their initial token lapse/expire). 

My initial thoughts right now is to do something like the following:

 - We implement token refresh: prior to the expiration of the access token, clients must use their regular access token to POST to a protected `/api/v1/refreshtoken` (or something) endpoint to get a new token (also with a 120 min expiration). 
 - Clients must keep getting new tokens while the user is active. 

This way if a user e.g. closes their laptop for a day, they must log in again. But if a user is active, they do no need to log in again. 

Another option is: we use `Flask-Session` and implement a server-side store of blacklisted tokens. Servers would issue clients an access token and a refresh token (a la OAuth2) and upon logout the refresh tokens get blacklisted. 

I'll read around a bit to see if there are any other common patterns we could use here prior to implementing something. Also curious what you think about this @emkll.

Reply:
I left some thoughts [here](https://github.com/freedomofpress/securedrop/pull/3619#discussion_r199971936) that address this.